engine/lib/tags.php

```diff
@@ -13,9 +13,9 @@
  *
  * @param string $string Comma-separated tag string
  *
- * @return array|false An array of strings, or false on failure
+ * @return array|false|mixed An array of strings, false if no data provided, or the original data if input was not a string
  */
-function string_to_tag_array($string) {
+function string_to_tag_array($string = false) {
```

Comment:
We should not make the argument optional, but I think we should rename it "tags". It's confusing to see a `return $string` for the case when we know it's not a string :)